app/src/main/java/com/lagradost/cloudstream3/ui/player/DownloadedPlayerActivity.kt

```diff
@@ -36,6 +39,22 @@ class DownloadedPlayerActivity : AppCompatActivity() {
     }
 
     private fun playLink(url: String) {
+        // check if url matches any provider
+        synchronized(APIHolder.apis) {
+            for (api in APIHolder.apis) {
+                if (url.startsWith(api.mainUrl) ||
+                    // this is to match mirror domains - like example.com, example.net
```

Comment:
I don't quite understand why you used a fuzzy match here. Most providers with mirror domains already handle mirror domains. This approach might cause a URL not to load that would have done so otherwise, for example if you have two different providers that have similar domains.